### ISSUE TITLE
Auto completion of commands and options?

### ISSUE THREAD

**Issue by janpio** [ionic-team/ionic-cli#2482]
Is it possible for a CLI tool to offer auto complete for commands and options? 

Right now this only works for file names in my command line, but that would be really handy if Ionic CLI could autocomplete `ionic c` to `ionic cordova` when I press [Tab].

**Comment by dwieeb** [comment]
Something I had hoped for, too. Shouldn't be too hard since we have the metadata of all the commands. Maybe in the future, sometime.

**Comment by janpio** [comment]
Some other CLIs seem to offer something like this:
https://akrabat.com/quick-tip-openwhisk-autocompletion/

**Comment by dwieeb** [comment]
@janpio Yep, we'd have to ship autocomplete functionality with the CLI. But we'd need to generate it from the command metadata just like how we generate the CLI docs. It would be a cool project but I don't have the time to prioritize it yet, unfortunately.

**Comment by dwieeb** [comment]
ZSH completion is going to be in CLI 5: https://github.com/ionic-team/ionic-cli/issues/3946

Still no updates for bash.
